src/catkin_pkg/workspaces.py

```diff
@@ -75,14 +75,21 @@ def order_paths(paths_to_order, prefix_paths):
     :param paths_to_order: list of paths
     :param prefix_paths: list of prefixes, must not end with '/'
     """
+    def contains(prefix, path):
```

Comment:
Please move this definition outside of the function and prefix it with an underscore. That way it can be accessed from the outside (e.g. from the test).

Also the order of the arguments seems unintuitive. I would expect the opposite order of arguments - reading: "path contains prefix".

Reply:
I guess `contains` meaning is ambiguous and depends on argument types:
- strings - `/etc/fstab` contains `/etc`
- paths (prefix understood as directory) - `/etc` contains `/etc/fstab`

It seems natural in Python to check containment by operator `in`, which uses container as rhs arg - e.g. `1 in [1]`.

I think `def _in_dir(path, dir_):` (then usage reads: `path in prefix, where prefix is dir`) would better explain its purpose, unless abbreviations in function names are frowned upon.

I noticed that private functions are usually placed immediately below functions which use them - I did the same.

Reply:
With Python 3 this becomes easier using `pathlib.Path.parents`. Maybe naming the function accordingly is even better: e.g. `_in_parents` which would map directly to what Python 3 code would express?

Reply:
I thought about that (or name mentioning "ancestor"), but both terms don't involve checking for identity. I.e. `/etc` is not among parents/ancestors of `/etc`
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents

Reply:
Since it is only an internal helper function maybe just very verbose: `_is_equal_or_parent`?

Reply:
I agree with compound name. However, `/etc` is not a parent of `/etc/apt/sources.list`, and I'd prefer to have something resembling container on rhs. Which brings us to: `_is_equal_or_descendant`. Does it sound ok?

Reply:
`_is_equal_or_in_parents`? Since in Python 3 would do: `p1 == p2 or p1 in p2.parents`

Reply:
Reasonable, I've done that.